### ISSUE TITLE
Building Static Problems

### ISSUE THREAD

**Issue by frink** [aperezdc/signify#19]
Sorry to report that I'm having issues building EXTRA_LDFLAGS='-static'...

It complains about pthread_mutex_lock and unlock as undefined references. I'm on Arch Linux with a fresh everything. libgcc_eh.c and arc4random.c in libbsd seem to be the offending files. These are definitely upstream files, but I can't verify whether the problem is in this projects Makefile or in that projects code. So there probably needs to be collaboration on this one like on ticket #16.

**Comment by frink** [comment]
NOTE: @guillemj @aperezdc - I've moved the report here from #16

**Comment by frink** [comment]
Digging deeper it appears that it's not including the pthread.h file from either glibc or musl depending on the MUSL flag. Since I'm most interested in a small build, I'll try digging deeper to find the issue with MUSL.

I'm thinking this is entirely a problem with makefiles...

**Comment by aperezdc** [comment]
@frink I have been able to produce static binaries using the following incantations, using Arch Linux:

```sh
# For a glibc-based build:
make EXTRA_LDFLAGS='-static -pthread' EXTRA_CFLAGS='-pthread' \
     BUNDLED_LIBBSD=1
# For a musl-based build:
make EXTRA_LDFLAGS='-static -pthread' EXTRA_CFLAGS='-pthread' \
     MUSL=1 CC=musl-gcc LD=musl-gcc
```

I also needed the small fix in commit 45874a3c0878a64cf63b144cbfe6a0a8853f8a66.

I'll try to add some logic the the `Makefile` to simplify making static builds.
